packages/angular-cli/plugins/karma.js

```diff
@@ -43,6 +44,25 @@ const init = (config) => {
     .filter((file) => config.preprocessors[file].indexOf('angular-cli') !== -1)
     .map((file) => config.preprocessors[file])
     .map((arr) => arr.splice(arr.indexOf('angular-cli'), 1, 'webpack', 'sourcemap'));
+
+  // Add global scripts
+  if (appConfig.scripts && appConfig.scripts.length > 0) {
+    const globalScriptPatterns = appConfig.scripts
+      .map(script => typeof script === 'string' ? { input: script } : script)
+      // Neither renamed nor lazy scripts are currently supported
+      .filter(script => !(script.output || script.lazy))
+      .map(script => ({
+        pattern: path.resolve(appRoot, script.input),
+        included: true,
+        served: true,
+        watched: true
+      }));
+
+    // Unshift elements onto the beginning of the files array.
+    // It's important to not replace the array, because
+    // karma already has a reference to the existing array.
+    Array.prototype.unshift.apply(config.files, globalScriptPatterns);
```

Comment:
We can't do that in the karma plugin it's not preprocessed and node4 does not support the spread operator (without a flag). 

I hope to tackle that in https://github.com/angular/angular-cli/issues/3605 by making the plugin be in TypeScript.